src/product.jl

```diff
@@ -60,12 +60,18 @@ function Base.convert(::Type{ProductSector{T}}, t::Tuple) where {T <: SectorTupl
     return ProductSector{T}(convert(T, t))
 end
 
-Base.one(::Type{ProductSector{T}}) where {I <: Sector, T <: Tuple{I}} = ProductSector((one(I),))
-function Base.one(::Type{ProductSector{T}}) where {I <: Sector, T <: Tuple{I, Vararg{Sector}}}
-    return one(I) ⊠ one(ProductSector{Base.tuple_type_tail(T)})
+function unit(::Type{ProductSector{T}}) where {I <: Sector, T <: Tuple{I, Vararg{Sector}}}
+    reduce(&, map(UnitStyle, _sectors(T))) == GenericUnit() &&
```

Comment:
I'm not sure if this works with any:
```suggestion
    any(c -> UnitStyle(c) == GenericUnit(), _sectors(T)) &&
```
alternatively:
```julia
    mapreduce(c -> UnitStyle(c) == GenericUnit(), |, _sectors(T); init=false) &&
```

Note also that I think reducing with `&` is not what you want, this is not defined if any of the "factors" has multiple units, not if all of them do.

Reply:
You're right, just one of them needs multiple units